cpp-rt.hpp

```diff
@@ -0,0 +1,100 @@
+#ifndef CPPRT_HPP
+#define CPPRT_HPP
+
+#include <stddef.h>
+#include <malloc.h>
+#include <cstdint>
+#include <cstdlib>
+#include <cassert>
+
+static inline void FATAL(const char* s) { fprintf(stderr, s); exit(1); }
+
+// This is a hack for .SO to work when they are compiled against a program/library that uses libstdc++
+// these stubs are emitted for default pure virtuals, they are present in the libstdc++ runtime. We will
+// have to redirect them to the phantom handler
+extern "C" {
+static inline __attribute__((noreturn)) void __phantom_handler(void) { fprintf(stderr, "a stub was not implemented!!!\n"); exit(1); }
```

Comment:
Might be nitpicky but do you think `abort` makes more sense here?

Reply:
Right, will move to abort, should fail with style